mountpoint-s3/tests/fuse_tests/cache_test.rs

```diff
@@ -288,6 +290,67 @@ where
     assert!(block.is_none());
 }
 
+/// A test that checks that data is not written to the bucket owned by an unexpected account
+#[test_case(get_express_bucket(), true, true; "bucket owned by the expected account")]
+#[test_case(get_external_express_bucket(), true, false; "bucket owned by another account")]
+#[test_case(get_external_express_bucket(), false, false; "bucket owned by another account, not checked")]
+#[cfg(feature = "s3express_tests")]
+fn express_cache_expected_bucket_owner(cache_bucket: String, owner_checked: bool, owner_matches: bool) {
+    use futures::executor::block_on;
+    use mountpoint_s3_client::config::S3ClientConfig;
+    use mountpoint_s3_client::error::ObjectClientError;
+    use mountpoint_s3_client::S3RequestError;
+
+    let bucket_owner = get_bucket_owner();
+    // Configure the client to enforce the bucket owner
+    let mut client_config = S3ClientConfig::default()
+        .part_size(CLIENT_PART_SIZE)
+        .endpoint_config(get_test_endpoint_config())
+        .read_backpressure(true)
+        .initial_read_window(CLIENT_PART_SIZE);
+    if owner_checked {
+        client_config = client_config.bucket_owner(&bucket_owner);
+    }
+    let client = S3CrtClient::new(client_config).unwrap();
+
+    // Create cache and mount a bucket
+    let bucket = get_standard_bucket();
+    let prefix = get_test_prefix("express_expected_bucket_owner");
+    let cache = ExpressDataCache::new(client.clone(), Default::default(), &bucket, &cache_bucket);
+    let cache_valid = block_on(cache.verify_cache_valid());
+    if owner_checked && !owner_matches {
+        match cache_valid {
+            Err(ObjectClientError::ClientError(S3RequestError::Forbidden(..))) => (),
+            _ => panic!("expected S3RequestError::Forbidden, got: {:?}", cache_valid),
+        }
+    } else {
+        cache_valid.unwrap(); // Ok(()) is expected
```

Comment:
nit: This is a use for `expect(..)`

```suggestion
        cache_valid.expect("should succeed if not enforcing bucket owner");
```